### PR TITLE
Feature/svg post [OFWEB-862]

### PR DESCRIPTION
**Pull request by SinisaG**
https://motainteam.atlassian.net/browse/OFWEB-862

### CODE REVIEW

File: _posts/2015-11-13-svg-sprites-angular.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
+---
+layout: post
+title: Svg sprites
+permalink: /svg-sprites
+tags: angular, svg, sprite
+author_id: sinisag
+category: webapp
+---
+
+(CSS) Sprite is a method, where we combine multiple small images or graphic in to one large image and access the needed graphic by specifying 'coordinates' on the big image.
```
Comment (maximevaly) [inline _posts/2015-11-13-svg-sprites-angular.md]: We could add a link to https://en.wikipedia.org/w/index.php?title=Sprite_(computer_graphics)&redirect=no#Sprites_by_CSS
Also "in to" -> "into"

File: _posts/2015-11-13-svg-sprites-angular.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+The main reason behind it is to improve performance of website. To understand this better, we need to be aware of browsers technical limitations.
+
+## Browser limitations
+
+- Limited amount of concurrent connections ([more] (http://www.browserscope.org/?category=network&v=top))
+- Limited amount of concurrent connections to the same hostname
+
+The second limitation we can tackle by serving assets from different hosts, but we are still limited with amount of concurrent connections.
```
Comment (maximevaly) [inline _posts/2015-11-13-svg-sprites-angular.md]: "limited with amount" -> "limited by the amount" ?

File: _posts/2015-11-13-svg-sprites-angular.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+In addition every http request comes with certain amount of overhead. We can partly address this by serving assets from cookie free domains (which should be the case regardless if you use sprites or not), but we can't fully prevent the overhead.
+
+So generating sprites became very popular way to overcome those limitations. With joining multiple icons/assets in to one big asset sheet, we download all related assets with one http request.
```
Comment (maximevaly) [inline _posts/2015-11-13-svg-sprites-angular.md]: became `a` very

File: _posts/2015-11-13-svg-sprites-angular.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+would cause an overhead. That is bad for multiple reasons, most obvious one being downloading more data then needed and therefore increase of download time.
+
+Ok..so I create 2 sprite maps and serve them based on device type? Well, is not that simple, what if your icons change color based on context they are displayed in? Very quickly, you can end up with numerous amount of sprite files and complex logic to deal with them. At this point things can get out of hand very quickly.
+
+## There is a better way!
+
+Meet [svgs] (https://sl.wikipedia.org/wiki/Scalable_Vector_Graphics)! Svg stands for Scalable Vector Graphic. Main advantage of svgs is that they are, as name suggests, scalable vector graphics. That means they will be sharp on all type of screens and in all sizes - you need just one asset to achieve that. But it gets better, you can also change icon color, width, height (and much more!) through css properties. That means, that if you can have your repeatable icons in form of svgs, then you have to make only one sprite and it will serve any purpose.
```
Comment (maximevaly) [inline _posts/2015-11-13-svg-sprites-angular.md]: I think the English version of the page would be clearer :D https://en.wikipedia.org/wiki/Scalable_Vector_Graphics

File: _posts/2015-11-13-svg-sprites-angular.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+## Generate svg sprite
+
+Here, we will assume, that you already have assets in svg format. You can take all your svgs and convert them to symbols. You can use a gulp/grunt (or any other task) to do it. Or if you prefer to manage you icons with external tools, you can use something like [fontastic] (http://fontastic.me).
+
+So let's assume we have 3 svg icons.
+
+facebook.svg
+
+```
+<svg width="30" height="30" viewBox="0 0 30 30" xmlns="http://www.w3.org/2000/svg" xmlns:sketch="http://www.bohemiancoding.com/sketch/ns"><title>social_facebook</title><desc>Created with Sketch.</desc><path d="M0 15c0 8.285 6.715 15 15 15s15-6.715 15-15c0-8.284-6.715-15-15-15s-15 6.716-15 15zm18.552-4.635h-2.254c-.268 0-.564.35-.564.819v1.629h2.819v2.321h-2.819v6.971h-2.662v-6.971h-2.414v-2.321h2.414v-1.366c0-1.959 1.36-3.553 3.226-3.553h2.254v2.471z" sketch:type="MSShapeGroup"/></svg>
```
Comment (maximevaly) [inline _posts/2015-11-13-svg-sprites-angular.md]: You can specify the syntax of the code block to apply nice colors (see the "Code" section https://guides.github.com/features/mastering-markdown/#examples)

``` SVG
<svg width="30" height="30" viewBox="0 0 30 30" xmlns="http://www.w3.org/2000/svg" xmlns:sketch="http://www.bohemiancoding.com/sketch/ns"><title>social_facebook</title><desc>Created with Sketch.</desc><path d="M0 15c0 8.285 6.715 15 15 15s15-6.715 15-15c0-8.284-6.715-15-15-15s-15 6.716-15 15zm18.552-4.635h-2.254c-.268 0-.564.35-.564.819v1.629h2.819v2.321h-2.819v6.971h-2.662v-6.971h-2.414v-2.321h2.414v-1.366c0-1.959 1.36-3.553 3.226-3.553h2.254v2.471z" sketch:type="MSShapeGroup"/></svg>
```

File: _posts/2015-11-13-svg-sprites-angular.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+<svg class="svg-icon">
+    <use xlink:href="#icon-social-facebook"></use>
+</svg>
+
+```
+
+We can apply some css properties to it (to svg-icon class) and style it the way we want and that would be it. With that little change, you can improve the way you handle your assets and make it much simpler to use in different context.
+
+
+## Gotchas - is not always roses
```
Comment (maximevaly) [inline _posts/2015-11-13-svg-sprites-angular.md]: Do you have any reference for these?

File: _posts/2015-11-13-svg-sprites-angular.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+
+Currently, we are embedding svg sprites in our template, just after opening `<body>` tag for good cross browser support. If you want to keep you resources out of your template, you can use a great shim written by Jonathan Neal [svg4everybody] (https://github.com/jonathantneal/svg4everybody).
+
+And if you need to support older browsers (IE8 etc.) you can use image as fallback.
+
+## Examples
+
+[Here] (https://github.com/SinisaG/svg-blog-demo) are a few examples that shows some of the browser glitches.
+
+You can run the examples using [5minutefork] (http://5minfork.com/). More in readme....
```
Comment (maximevaly) [inline _posts/2015-11-13-svg-sprites-angular.md]: :+1: You can also link directly to the fork with http://5minfork.com/SinisaG/svg-blog-demo

File: _posts/2015-11-13-svg-sprites-angular.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+
+## Base tag
+
+If you are like us and use [AngularJs] (https://angularjs.org/) everything written above will apply if you are using urls with hashbang (#!). But if you are using pretty urls with `<base>` tag or you use this tag for any other reason, then be aware, that firefox is interpreting this differently then other browsers, so you will need to reference svgs with an absolute url. For angular or similar environments we suggest using a directive that keeps urls up to date:
+
+
+```javascript
```
Comment (maximevaly) [inline _posts/2015-11-13-svg-sprites-angular.md]: Isn't that something that we could / should add to https://github.com/Onefootball/onefootball-angular-components ?

File: _posts/2015-11-13-svg-sprites-angular.md [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+
+```
+<svg class="icon-nav-info">
+    <use of-svg-icon icon="'#icon-nav-admin'"></use>
+</svg>
+```
+
+That has proven to work for us in all browsers we tend to support. Live example - [Onefootball - best app for football] (https://www.onefootball.com) ;)
+
+I hope this clears sprites and svg sprites for you. Happy performance improving!
```
Comment (maximevaly) [inline _posts/2015-11-13-svg-sprites-angular.md]: Maybe we could clonclude by listing some resources about SVG sprites, like Sara Soueidan's articles https://24ways.org/2014/an-overview-of-svg-sprite-creation-techniques/ & http://sarasoueidan.com/blog/svg-tips-for-designers/, or that SVG sandbox https://github.com/metamn/svg, or talk about other uses of SVG like fonts http://caniuse.com/#feat=svg-fonts

**Comment by maximevaly** [comment]
Looks good, I'd just do a quick search-and-replace to add a few capital letters on acronyms and names like in CSS, Facebook, Google, Firefox, SVG, iOS, AngularJS....

File: _posts/2015-11-13-svg-sprites-angular.md (hunk 4 above)
Comment (SinisaG) [inline _posts/2015-11-13-svg-sprites-angular.md]: Not to our readers in Slovenia! :D Thanks. :)

File: _posts/2015-11-13-svg-sprites-angular.md (hunk 6 above)
Comment (SinisaG) [inline _posts/2015-11-13-svg-sprites-angular.md]: Unfortunatly no. Is from what we learned by using it.

File: _posts/2015-11-13-svg-sprites-angular.md (hunk 7 above)
Comment (SinisaG) [inline _posts/2015-11-13-svg-sprites-angular.md]: Much nicer! Thanks.

File: _posts/2015-11-13-svg-sprites-angular.md (hunk 8 above)
Comment (SinisaG) [inline _posts/2015-11-13-svg-sprites-angular.md]: https://motainteam.atlassian.net/browse/OFWEB-723 :)

**Comment by SinisaG** [comment]
Updated.

**Comment by SinisaG** [comment]
Updated + examples updated.

**Pull request merged**